mantle/platform/qemu.go

```diff
@@ -306,6 +306,61 @@ func (inst *QemuInstance) SwitchBootOrder() (err error) {
 	return nil
 }
 
+// RemovePrimaryBlockDevice deletes the primary device from a qemu instance
+// and sets the seconday device as primary. It expects that all block devices
+// are mirrors.
+func (inst *QemuInstance) RemovePrimaryBlockDevice() error {
+	var primaryDevice string
+	var secondaryDevicePath string
+	monitor, err := newQMPMonitor(inst.tempdir)
+	if err != nil {
+		return errors.Wrapf(err, "Could not connect to QMP device")
+	}
+	if err := monitor.Connect(); err != nil {
+		return errors.Wrapf(err, "Could not connect to QMP device")
+	}
+
+	defer func() {
+		e := monitor.Disconnect()
+		if err != nil {
+			err = fmt.Errorf("%v; %v", err, e)
+		} else {
+			err = e
+		}
+	}()
+	blkdevs, err := listQMPBlkDevices(monitor, inst.tempdir)
+	if err != nil {
+		return errors.Wrapf(err, "Could not list block devices through qmp")
+	}
+	for _, dev := range blkdevs.Return {
+		if !dev.Removable {
+			if dev.Inserted.BackingfileDepth == 1 {
+				primaryDevice = dev.DevicePath
+			} else {
+				secondaryDevicePath = dev.DevicePath
+			}
+		}
+	}
+	err = setBootIndexForDevice(monitor, primaryDevice, -1)
+	if err != nil {
+		return errors.Wrapf(err, "Could not set bootindex for %v", primaryDevice)
+	}
+	primaryDevice = primaryDevice[:strings.LastIndex(primaryDevice, "/")]
```

Comment:
I'm not sure which (or maybe both are) is correct but when I run boot mirror tests locally I always use `Device` instead of `DevicePath` when running `drive_del` in QEMU Monitor. Notably I also haven't ran into problems when running on UEFI doing that. Probably worth looking into more but might be worth giving it a shot.

That said I do still use `DevicePath` when running `qom-set` so you'd need to store it in an additional variable if you wanted to move to that method.

Reply:
IIUC, `drive_del` won't accept `devicePath` as input since it requires the device itself i.e. `d1,d2` etc
However, for `device-del`, you'd normally provide the devicePath (https://qemu.readthedocs.io/en/latest/system/monitor.html)